### ISSUE TITLE
IfcConvert parallel geometry interpretation results in missing elements

### ISSUE THREAD

**Issue by eriadam** [IfcOpenShell/IfcOpenShell#1026]
Hello,

We noticed that running `IfcConvert` with parallel geometry interpretation enabled (eg: `-j2`) results in some elements/meshes missing from the output `dae` (and `glb`) files.

Enclosed is the two output from the same model.

[Archive.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/5317578/Archive.zip)

Thanks,
Adam

**Comment by aothms** [comment]
Yes this is probably the same as https://github.com/IfcOpenShell/IfcOpenShell/issues/724#issuecomment-554742981

Shape reuse is not valid and it trips up the multiprocessing

```
[Warning] Representation used by IfcRepresentationMap and IfcProductDefinitionShape
#803530=IfcShapeRepresentation(#38786,'FootPrint','GeometricCurveSet',(#803528))
```

Are you experiencing this on actual end-user data, or only on this test model?

**Comment by eriadam** [comment]
It is with both test files and end-user files.

**Comment by Amoki** [comment]
We also noticed this with end-user files.

**Comment by BIMvoice** [comment]
Retested the sample on the current v0.8 line (OCC 7.9.2). Converting single-threaded and multi-threaded now yields the same element count (1610 mesh groups with -j1 and with -j2), so the parallel path no longer drops elements. The multithreaded iterator was reworked to a deterministic reordering buffer keyed on task ordinal. Recommend closing as fixed unless you still see missing elements on a current build.
